x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rest/RestGetInferenceModelAction.java

```diff
@@ -26,13 +26,21 @@ public String getName() {
 
     @Override
     public List<Route> routes() {
-        return List.of(new Route(GET, "_inference/{task_type}/{model_id}"));
+        return List.of(new Route(GET, "_inference/{task_type}/{model_id}"), new Route(GET, "_inference/_all"));
     }
 
     @Override
-    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
-        String taskType = restRequest.param("task_type");
-        String modelId = restRequest.param("model_id");
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) {
+        String modelId = null;
+        TaskType taskType = null;
+        if (restRequest.hasParam("task_type") == false && restRequest.hasParam("model_id") == false) {
+            // _all models request
+            modelId = "_all";
+            taskType = TaskType.ANY;
+        } else {
+            taskType = TaskType.fromStringOrStatusException(restRequest.param("task_type"));
```

Comment:
Right now, the internal-only task type `ANY` leaks to the outside world, right?

It's not really a problem here, but it's also not particularly nice, and it's caused by hacking `ANY` into the TaskType enum.

Reply:
E.g. I think that `/_inference/any/elser` gives

```
Requested task type [ANY] does not match the model's task type [TASKTYPE]
```